rust/toolchain.bzl

```diff
@@ -628,6 +640,11 @@ rust_toolchain = rule(
             doc = "The operating system for the current toolchain",
             mandatory = True,
         ),
+        "rust_analyzer_proc_macro_srv": attr.label(
```

Comment:
My two cents about this attribute is it should either be on `rust_analyzer_toolchain` or maybe it doesn't need to exist at all. This change seems to add the ability to track the binary as part of the toolchain but it's ultimately not used by Bazel? Rust-analyzer works outside of Bazel so as long as the sources exist on disk in your output user root, the binary would be usable? Are any of my questions correct here? 😅

Reply:
IIUC the whole idea of this new binary is that it should track the toolchain used to compile the proc_macro dylibs so that RA doesn't break if the proc_macro ABI changes between Rust versions (e.g. you use nightly RA but are building an older stable version).

With that, my thinking was that we want to make sure that when we construct the rust-project.json file the sysroot specified will point to a proc-macro-srv that matches the Rust toolchain used to compile the proc macro dylibs, which lead me to this where we wire up the proc-macro-srv present in the Rust toolchain being used. It's true that we don't need to place the proc-macro-srv in execroot for any of this to work if we had the true sysroot, but it wasn't clear to me how I'd extract the real location of the toolchain sysroot from within Skylark, though maybe there is a way? 

It might be possible to get this working by putting it on the RA toolchain, but to me it seemed like it would then be defined independent of the Rust toolchain and thus not linked to the rustc version used to compile the proc_macro dylibs?

Reply:
I think it's fair to point rust_analyzer at a sysroot but unfortunately in Bazel it's not quite the same as for Cargo. The sysroot would likely be a `rust_toolchain_repository` which only contains components used for building on one exec platform to one target platform. The sysroot would change depending on what you were targeting. Would that be an issue for `rust-analyzer-proc-macro-svr`? It kinda seems like it wouldn't be given the binary comes with the `rustc` components which would largely be same across all these sysroots.